signature-aggregator/aggregator/cache/cache.go

```diff
@@ -0,0 +1,69 @@
+package cache
+
+import (
+	"math"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/utils/crypto/bls"
+	"github.com/ava-labs/avalanchego/utils/logging"
+	lru "github.com/hashicorp/golang-lru/v2"
+	"github.com/pingcap/errors"
+	"go.uber.org/zap"
+)
+
+type Cache struct {
+	logger     logging.Logger
+	signatures *lru.Cache[cacheKey, cacheValue]
+}
+
+type cacheKey ids.ID // warp message ID
+type cacheValue map[PublicKeyBytes]SignatureBytes
```

Comment:
Can we just use the regular types directly? It seems odd to have to wrap everything before passing them in

Reply:
For the public keys, we _can_ use the regular types directly in `cache.go` but when the interactions happen from `aggregator.go` they will need to be wrapped in _something_ either way, because `warp.Validator.PublicKeyBytes` is a slice, and a map key can only be a fixed length array, so even if we change to using regular types directly, we'll still have to wrap the arguments to `Get` and `Add` with `[bls.PublicKeyLen]byte(...)`.

I'm fine with going either way on it, but at the time I wrote this it felt "cleaner" to have a name for a type that needs to be repeated several times over. (Note this is just like what's done with the `type blsSignatureBuf [bls.SignatureLen]byte` in `aggregator.go`.)

Reply:
I think we could pass in `thisIsASlice[:]' to a function that takes a fixed size array to get the conversion

Reply:
I tried, without any luck:
```
gene-dev 08/13 22:21:21 (signature-aggregation-api-cache) 1 ~/dev/awm-relayer$ git diff signature-aggregator/                                                   
diff --git a/signature-aggregator/aggregator/aggregator.go b/signature-aggregator/aggregator/aggregator.go                                                      
index d500e82..73ee1da 100644                                                                                                                                   
--- a/signature-aggregator/aggregator/aggregator.go                                                                                                             
+++ b/signature-aggregator/aggregator/aggregator.go                                                                                                             
@@ -412,7 +412,7 @@ func (s *SignatureAggregator) handleResponse(                                                                                               
                signatureMap[vdrIndex] = signature                                                                                                              
                s.cache.Add(                                                                                                                                    
                        unsignedMessage.ID(),                                                                                                                   
-                       cache.PublicKeyBytes(validator.PublicKeyBytes),                                                                                         
+                       validator.PublicKeyBytes[:],                                                                                                            
                        cache.SignatureBytes(signature),                                                                                                        
                )                                                                                                                                               
                accumulatedSignatureWeight.Add(accumulatedSignatureWeight, new(big.Int).SetUint64(validator.Weight))                                            
diff --git a/signature-aggregator/aggregator/cache/cache.go b/signature-aggregator/aggregator/cache/cache.go                                                    
index e0fa57d..6d7a4e8 100644                                                                                                                                   
--- a/signature-aggregator/aggregator/cache/cache.go                                                                                                            
+++ b/signature-aggregator/aggregator/cache/cache.go                                                                                                            
@@ -52,7 +52,7 @@ func (c *Cache) Get(msgID ids.ID) (cacheValue, bool) {                                                                                        
                                                                                                                                                                
 func (c *Cache) Add(                                                                                                                                           
        msgID ids.ID,                                                                                                                                           
-       pubKey PublicKeyBytes,                                                                                                                                  
+       pubKey [bls.PublicKeyLen]byte,                                                                                                                          
        signature SignatureBytes,                                                                                                                               
 ) {                                                                                                                                                            
        var (                                                                                                                                                   
gene-dev 08/13 22:21:27 (signature-aggregation-api-cache) 0 ~/dev/awm-relayer$ scripts/build.sh                                                                 
Running protobuf fmt...                                                                                                                                         
Running protobuf lint check...                                                                                                                                  
Re-generating protobuf...                                                                                                                                       
Building AWM Relayer Version: v1.4.0-rc.2 at /home/gene/dev/awm-relayer/build/awm-relayer                                                                       
# github.com/ava-labs/awm-relayer/signature-aggregator/aggregator                                                                                               
signature-aggregator/aggregator/aggregator.go:415:28: cannot use validator.PublicKeyBytes[:] (value of type []byte) as [48]byte value in argument to s.cache.Add
Building Signature Aggregator Version: v1.4.0-rc.2 at /home/gene/dev/awm-relayer/signature-aggregator/build/signature-aggregator                                
# github.com/ava-labs/awm-relayer/signature-aggregator/aggregator                                                                                               
aggregator/aggregator.go:415:28: cannot use validator.PublicKeyBytes[:] (value of type []byte) as [48]byte value in argument to s.cache.Add                     
```

Feel free to give it a try and let me know what you prefer.

Reply:
addressed in https://github.com/ava-labs/awm-relayer/pull/415/commits/aa631b608d8c85e6c1a3e23815d0cf568f608184